packages/sanity/src/core/form/inputs/ReferenceInput/PreviewReferenceValue.tsx

```diff
@@ -57,21 +59,94 @@ export function PreviewReferenceValue(props: {
     }
   }
 
+  const availability = referenceInfo.result.availability
+
+  const notFound = availability.reason === 'NOT_FOUND'
+  const insufficientPermissions = availability.reason === 'PERMISSION_DENIED'
+  if (insufficientPermissions || notFound) {
+    return (
+      <Inline space={2}>
+        <Box padding={1}>
+          <Flex align="center">
+            <Box flex={1} paddingY={2}>
+              <Text muted>Document unavailable</Text>
+            </Box>
+          </Flex>
+        </Box>
+        {insufficientPermissions || notFound ? (
+          <Box>
+            <Tooltip
+              portal
+              content={
+                notFound ? (
+                  <UnavailableMessage title="Not found" icon={HelpCircleIcon}>
+                    The referenced document does not exist
+                    <br />
+                    (id: <code>{value._ref}</code>)
+                  </UnavailableMessage>
+                ) : (
+                  <UnavailableMessage title="Insufficient permissions" icon={AccessDeniedIcon}>
+                    The referenced document could not be accessed due to insufficient permissions
+                  </UnavailableMessage>
+                )
+              }
+            >
+              <TextWithTone tone="default">
+                <HelpCircleIcon />
+              </TextWithTone>
+            </Tooltip>
+          </Box>
+        ) : null}
+      </Inline>
+    )
+  }
+
   const refTypeName = referenceInfo.result?.type
   const refType = type.to.find((toType) => toType.name === refTypeName)
 
   if (!refType) {
     return (
-      <Stack space={2} padding={2}>
-        The referenced document is of invalid type: ({refTypeName || 'unknown'})
-        <pre>{JSON.stringify(value, null, 2)}</pre>
-      </Stack>
+      <Flex align="center" justify="flex-start">
+        <Box padding={1}>
+          <Flex align="center">
+            <Box flex={1} paddingY={2}>
+              <Text muted>Type of referenced document not declared in schema</Text>
```

Comment:
I would suggest to change this copy to "Document of invalid type"

Reply:
Agreed - fixed by 544c368153